proof/archive_test.go

```diff
@@ -56,74 +112,101 @@ func TestFileArchiver(t *testing.T) {
 
 		fetchFunc func(*FileArchiver) error
 
-		expectedErorr error
+		expectedFetchError error
+		expectedStoreError error
 	}{
 		// Attempting to fetch a proof that doesn't exist on disk should
 		// return an error.
 		{
 			name: "proof not found",
 			locator: Locator{
-				AssetID:   randAssetID(t),
+				AssetID:   randAssetID(),
 				ScriptKey: *test.RandPubKey(t),
 			},
-			expectedErorr: ErrProofNotFound,
+			expectedFetchError: ErrProofNotFound,
 		},
 
-		// Attempting to fetch a file on disk that doesn't have an asset ID
-		// specified should return an error.
+		// Attempting to fetch a file on disk that doesn't have an asset
+		// ID specified should return an error.
 		{
 			name: "invalid asset ID",
 			locator: Locator{
 				ScriptKey: *test.RandPubKey(t),
 			},
-			expectedErorr: ErrInvalidLocatorID,
+			expectedFetchError: ErrInvalidLocatorID,
 		},
 
 		// Fetching w/ the assetID, but not script key should return an
 		// error as well.
 		{
 			name: "invalid script key",
 			locator: Locator{
-				AssetID: randAssetID(t),
+				AssetID: randAssetID(),
 			},
-			expectedErorr: ErrInvalidLocatorKey,
+			expectedFetchError: ErrInvalidLocatorKey,
+		},
+
+		// Storing a proof with assetID and script key, but no outpoint
+		// should return an error as well.
+		{
+			name: "invalid outpoint",
+			locator: Locator{
+				AssetID:   randAssetID(),
+				ScriptKey: *test.RandPubKey(t),
+			},
+			proofBlob: func() Blob {
+				return bytes.Repeat([]byte{0x01}, 100)
+			},
+			expectedStoreError: ErrOutPointMissing,
 		},
 
 		// We should be able to insert a proof, then get it right back
 		// the same way we found it.
 		{
 			name: "proof happy path",
 			locator: Locator{
-				AssetID:   randAssetID(t),
+				AssetID:   randAssetID(),
 				ScriptKey: *test.RandPubKey(t),
+				OutPoint:  &wire.OutPoint{},
```

Comment:
If we add a slightly diff case (fetch proof, get annotated proof, assert locator), it should catch the issue mentioned above re not being able to recover the full outpoint from the new file prefix.